Corona-Warn-App/src/main/java/de/rki/coronawarnapp/vaccination/core/repository/storage/ValueSetsStorage.kt

```diff
@@ -26,72 +23,28 @@ class ValueSetsStorage @Inject constructor(
         return when (valueSetString != null) {
-            true -> gson.fromJson(valueSetString, StoredVaccinationValueSet::class.java)
-            else -> emptyVaccinationValueSet
-        }.also { loaded -> Timber.v("Loaded value set %s", loaded) }
+            true -> gson.fromJson(valueSetString)
+            else -> emptyValueSetsContainer
+        }.also { loaded -> Timber.v("Loaded value sets container %s", loaded) }
     }
 
-    private fun setValueSet(value: VaccinationValueSet) {
-        Timber.v("Saving value set %s", value)
+    private fun setValueSet(value: ValueSetsContainer) {
+        Timber.v("Saving value sets container %s", value)
         prefs.edit {
-            val json = gson.toJson(value.toStoredVaccinationValueSet(), StoredVaccinationValueSet::class.java)
+            val json = gson.toJson(value, ValueSetsContainer::class.java)
             Timber.v("Writing %s to prefs", json)
-            putString(PKEY_VALUE_SETS_PREFIX, json)
-        }
-    }
-
-    @VisibleForTesting
-    fun VaccinationValueSet.toStoredVaccinationValueSet(): StoredVaccinationValueSet =
-        StoredVaccinationValueSet(
-            languageCode = languageCode,
-            vp = vp.toStoredValueSet(),
-            mp = mp.toStoredValueSet(),
-            ma = ma.toStoredValueSet()
-        )
-
-    @VisibleForTesting
-    fun VaccinationValueSet.ValueSet.toStoredValueSet(): StoredVaccinationValueSet.StoredValueSet =
-        StoredVaccinationValueSet.StoredValueSet(
-            items = items.map { it.toStoredItem() }
-        )
-
-    @VisibleForTesting
-    fun VaccinationValueSet.ValueSet.Item.toStoredItem():
-        StoredVaccinationValueSet.StoredValueSet.StoredItem =
-            StoredVaccinationValueSet.StoredValueSet.StoredItem(
-                key = key,
-                displayText = displayText
-            )
-
-    @VisibleForTesting
-    data class StoredVaccinationValueSet(
-        @SerializedName("languageCode") override val languageCode: Locale,
-        @SerializedName("vp") override val vp: StoredValueSet,
-        @SerializedName("mp") override val mp: StoredValueSet,
-        @SerializedName("ma") override val ma: StoredValueSet
-    ) : VaccinationValueSet {
-
-        @Keep
-        data class StoredValueSet(
-            @SerializedName("items") override val items: List<StoredItem>
-        ) : VaccinationValueSet.ValueSet {
-
-            @Keep
-            data class StoredItem(
-                @SerializedName("key") override val key: String,
-                @SerializedName("displayText") override val displayText: String
-            ) : VaccinationValueSet.ValueSet.Item
+            putString(PKEY_VALUE_SETS_CONTAINER_PREFIX, json)
         }
     }
 }
 
 private const val PREF_NAME = "valuesets_localdata"
-private const val PKEY_VALUE_SETS_PREFIX = "valueset"
+private const val PKEY_VALUE_SETS_CONTAINER_PREFIX = "valuesets_container"
```

Comment:
what about saved keys already?

Reply:
Nobody will ever notice them

Reply:
what do you mean by that ? will is cause an issue or not ?

Reply:
It will not cause any issue.
But I can add a check and delete them if you want

Reply:
Alright, if you think , it won’t cause a problem, I'm OK , thanks for clarifying